_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,310 @@
+Traditionally, these tasks have been performed manually, often leading to errors, inconsistencies, and operational overhead.
+
+This is where the Strimzi Topic Operator comes into play.
+It introduces a declarative approach to Kafka topic management, focusing on the "what" rather than the "how". 
+In a declarative model, administrators define the desired state of topics, specifying high-level characteristics and requirements, while leaving the intricate details to the operator itself.
+This shift in perspective not only simplifies the management process but also ensures that the desired configuration is maintained consistently across the Kafka cluster.
+
+<figure>
+    <img src="/assets/images/posts/2023-10-30-uto-interactions.png" height=440>
+    <figcaption><small>Fig 1. Topic Operator interactions.</small></figcaption>
+</figure>
+
+Declarative topic management involves defining the desired state of a Kafka topic using custom resources.
+In particular, you use Kubernetes custom resources to declare how you want your Kafka topics to be configured and managed.
+
+Here's how a Kafka topic is declared using this approach:
+
+```sh
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 3
+  replicas: 2
+  config:
+    retention.ms: 86400000
+```
+
+The requirement here is that a single `KafkaTopic` custom resource is used to manage a single topic in a single Kafka cluster.
+From the example above, you can see that the resource has a label `strimzi.io/cluster` with the name of the target cluster.
+The operator communicates with the Kafka cluster and ensures that the specified Kafka topic is created or updated according to the desired state.
+It can be deployed standalone, or along with your Kafka cluster using the `Kafka` custom resource.
+
+The first generation of the Topic Operator is based on the Kafka Streams library, and uses ZooKeeper to gather topic metadata.
+We call it **Bidirectional Topic Operator (BTO)**, as it allows to reconcile changes coming from both Kubernetes and Kafka using a three-way merge logic.
+In order to do this, it has to keep an internal topic store, which is the source-of-truth used to determine the necessary changes for synchronization.
+If the Kafka-side configuration is changed out-of-band (e.g. by using the Admin client, or Kafka shell scripts), those changes will eventually be reverted.
+
+The new generation of the Topic Operator is called **Unidirectional Topic Operator (UTO)**.
+The UTO improves on scalability and maintenance, but also provides support for Kafka clusters running in KRaft mode.
+It is fully backwards compatible with the BTO, but it will only reconcile topic events in one direction, from Kubernetes to Kafka.
+Your `KafkaTopic` custom resources will be the source-of-truth, like with all the other operators.
```

Comment:
```suggestion
The **Unidirectional Topic Operator (UTO)** is the latest generation of Topic Operator, which improves on the BTO in terms of topic maintenance and scalability, but also provides support for Kafka clusters running in KRaft mode.
As the UTO only reconciles topic events from Kubernetes to Kafka,`KafkaTopic` custom resources become the single source-of-truth.
```

Reply:
We introduced the UTO once. Maybe we can reduce as suggested here^  I thought we could introduce maintenance and scalability here and then have separate subsections...